docs/advanced.rst

```diff
@@ -1018,105 +1136,665 @@ to learn more about the use of this feature.
+    :ref:`Discovery protocol <DS_discovery_protocol>`, how to make a participant a *client* or a *server*.
+    :ref:`Server unique id <DS_guidPrefx>`, how to link a *clients* to *servers*.
+    :ref:`Seting up transport <DS_locators>`, how to specify which transport to use and make *servers* reachable.
+    :ref:`Pinging period <DS_ping_period>`, how to fine tune server-client handshake.
+    :ref:`Matching period <DS_match_period>`, how to fine tune server deliver efficiency.
+
+.. _DS_discovery_protocol:
+
+Choosing between client and server
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+It is set by the :ref:`Discovery Protocol <discovery_protocol>` general attribute.
+A participant can only play a role (despite the fact that a *server* may act as a *client* of other server).
+It is mandatory to fill this value because it defaults to *simple*.
+The values associated with the Server-client discovery are specified in :ref:`discovery settings
+section <DS_DiscoverySettings>`.
+In the examples bellow is shown how to manage the corresponding enum attribute:
+
+.. code-block:: bash
+
+    ParticipantAttributes.rtps.builtin.discovery_config.discoveryProtocol
+
+and xml tag:
+
+.. code-block:: bash
+
+    dds>profiles>participant>rtps>builtin>discovery_config>discoveryProtocol
+
++------------------------------------------------------------+
+| **C++**                                                    |
++------------------------------------------------------------+
+| .. literalinclude:: ../code/CodeTester.cpp                 |
+|    :language: c++                                          |
+|    :start-after: //CONF_SERVER_DISCOVERY_PROTOCOL          |
+|    :end-before: //!--                                      |
++------------------------------------------------------------+
+| **XML**                                                    |
++------------------------------------------------------------+
+| .. literalinclude:: ../code/XMLTester.xml                  |
+|    :language: xml                                          |
+|    :start-after: <!-->CONF-SERVER-DISCOVERY-PROTOCOL<-->   |
+|    :end-before: <!--><-->                                  |
++------------------------------------------------------------+
+
+.. _DS_guidPrefx:
+
+The server unique identifier ``GuidPrefix``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This belongs to the RTPS specification and univocally identifies each DDS participant. It consist on 12 bytes and is
+basically a key in the DDS domain.
+In the server-client discovery has the purpose to link a *server* to its *clients*.
+Note that there is an auxiliary **ReadguidPrefix** method to populate the ``GuidPrefix`` using a ``string``.
+It must be mandatorily specified in:
+
+Server side setup
+"""""""""""""""""
+
+In the examples below we show how to setup the attribute:
+
+.. code-block:: bash
+
+    ParticipantAttributes.rtps.prefix
+
+and xml tag:
+
+.. code-block:: bash
+
+    dds>profiles>participant>rtps>prefix
```

Comment:
Remove the enum and tag hierarchy (it is implicit in the examples).

Reply:
As above... It's not easy to imply the jigsaw puzzle that the attributes are, directly, from the examples...that's the main reason I put these clues there.

Reply:
Same changes as above